### PR TITLE
[tests-only][full-ci]Add spaces tests for apiWebdavMove2 suite

### LINKED ISSUE #3107: update `apiWebdavMove2` suite for spaces webdav
### Description
- add spaces WebDAV URL for `apiWebdavMove2` suite from `oc/core`
- includes:
  - moveFile.feature
  - moveFileToBlacklistedName.feature
  - moveFileToExcludedDirectory.feature

### Related
- part of https://github.com/owncloud/ocis/issues/2854

### PR DESCRIPTION
**Pull request by amrita-shrestha**
## Description
adds spaces DAV path on `apiWebdavMove2 ` suite
- moves two test scenario related to moving folder from `apiWebdavMove2/moveFile.feature` to  `apiWebdavMove1/moveFolder.feature`
- adds new feature file` moveShareOnOcisfile` for testing Share related test only on ocis 

## Related Issue

- fixes https://github.com/owncloud/ocis/issues/3107

## How Has This Been Tested?

- Locally
- CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require an increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 
- [ ] Changelog item, see [TEMPLATE](https://github.com/owncloud/core/blob/master/changelog/TEMPLATE)

### CODE REVIEW

File: tests/TestHelpers/WebDavHelper.php [hunk 1]
```diff
@@ -80,7 +82,8 @@ public static function getFileIdForPath(
 			$path,
 			null,
 			$xRequestId,
-			$body
+			$body,
+			$davPathVersionToUse
```
Comment (kiranparajuli589) [inline tests/TestHelpers/WebDavHelper.php:86]: can we directly use the `$this->getDavPathVersion()` function here?
it should also give a new DAV path as default.

**Review by kiranparajuli589 — APPROVED** [review]

File: tests/TestHelpers/WebDavHelper.php (hunk 1 above)
Comment (kiranparajuli589) [inline tests/TestHelpers/WebDavHelper.php:86]: can we use the `$this->getDavPathVersion()` fn directly here?

**Comment by kiranparajuli589** [comment]
approved accidentally...sorry for the inconvenience!

File: tests/TestHelpers/WebDavHelper.php (hunk 1 above)
Comment (amrita-shrestha) [reply tests/TestHelpers/WebDavHelper.php:86]: $this->getDAVPathVersion() is in bootstrap folder .Upto now we didn't call bootstrap fuction from helper folder.But their is a way in php to do so.

File: tests/acceptance/features/apiWebdavMove2/moveFile.feature [hunk 2]
```diff
@@ -124,55 +151,7 @@ Feature: move (rename) file
       | old         | Brian |
       | new         | Brian |
 
-  @files_sharing-app-required
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveFile.feature:127]: hmm... :fishing_pole_and_fish:  folder test was in files feature :+1:

File: tests/acceptance/features/apiWebdavMove2/moveFile.feature [hunk 3]
```diff
@@ -307,17 +309,34 @@ Feature: move (rename) file
...
+    And the content of file "/#oc ab?cd=ef#" for user "Alice" should be "ownCloud test text file 0"
+    Examples:
+      | dav_version |
+      | old         |
+      | new         |
+
+    @skipOnOcV10 @personalSpace
     Examples:
-      | renamed_file  |
-      | #oc ab?cd=ef# |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveFile.feature:320]: this is also fine I think. if more example needed to be added for files, then another column can be easily added

File: tests/acceptance/features/apiWebdavMove2/moveFile.feature [hunk 4]
```diff
@@ -364,6 +398,11 @@ Feature: move (rename) file
       | old         |
       | new         |
 
+    @skipOnOcV10 @personalSpace
+    Examples:
+      | dav_version |
+      | spaces      |
+
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveFile.feature:405]: \n

File: tests/acceptance/features/apiWebdavMove2/moveFile.feature [hunk 5]
```diff
@@ -385,6 +424,11 @@ Feature: move (rename) file
...
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveFile.feature:431]: \n

File: tests/acceptance/features/apiWebdavMove2/moveFile.feature [hunk 6]
```diff
@@ -405,6 +449,11 @@ Feature: move (rename) file
...
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveFile.feature:456]: \n

File: tests/acceptance/features/apiWebdavMove2/moveFileToExcludedDirectory.feature [hunk 7]
```diff
@@ -19,6 +20,11 @@ Feature: users cannot move (rename) a file to or into an excluded directory
       | old         |
       | new         |
 
+    @skipOnOcV10 @personalSpace
+    Examples:
+      | dav_version |
+      | spaces      |
+
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveFileToExcludedDirectory.feature:27]: \n

File: tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature [hunk 8]
```diff
@@ -0,0 +1,217 @@
+@api @files_sharing-app-required @skipOnOcV10
+Feature: move (rename) file
+  As a user
+  I want to be able to move and rename files
+  So that I can manage my file system
+
+  Background:
+    Given using OCS API version "1"
+    And user "Alice" has been created with default attributes and without skeleton files
+    And parameter "shareapi_auto_accept_share" of app "core" has been set to "yes"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature:10]: this step is not necessary for ocis tests.

File: tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature [hunk 9]
```diff
@@ -0,0 +1,217 @@
...
+    And user "<mover>" has created folder "/testsubfolder"
+    And user "<mover>" has uploaded file with content "test data" to "/testsubfolder/testfile.txt"
+    When user "Alice" accepts share "/testshare" offered by user "Brian" using the sharing API
+    And user "<mover>" moves folder "/testsubfolder" to "<dst-folder>/testsubfolder" using the WebDAV API
+    Then the HTTP status code should be "201"
+    And the content of file "/Shares/testshare/testsubfolder/testfile.txt" for user "Alice" should be "test data"
+    And the content of file "/testshare/testsubfolder/testfile.txt" for user "Brian" should be "test data"
+    And as "<mover>" file "/testsubfolder" should not exist
+    Examples:
+      | dav_version | mover | dst-folder        |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature:91]:
```suggestion
      | dav_version | mover | destination-folder        |
```

Comment (kiranparajuli589) [reply tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature:91]: `dst` :face_with_spiral_eyes:

File: tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature [hunk 10]
```diff
@@ -0,0 +1,217 @@
...
+      | permissions | change    |
+      | shareWith   | Alice     |
+    When user "Alice" accepts share "/testshare" offered by user "Brian" using the sharing API
+    And user "<mover>" moves folder "<src-folder>/testsubfolder" to "/testsubfolder" using the WebDAV API
+    Then the HTTP status code should be "201"
+    And the content of file "/testsubfolder/testfile.txt" for user "<mover>" should be "test data"
+    And as "Alice" folder "<src-folder>/testsubfolder" should not exist
+    And as "Brian" folder "/testshare/testsubfolder" should not exist
+    Examples:
+      | dav_version | mover | src-folder        |
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature:122]:
```suggestion
      | dav_version | mover | source-folder        |
```

Comment (kiranparajuli589) [reply tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature:122]: use the same separators, `-` or `_`
also, look for other occurrences in the file

File: tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature [hunk 11]
```diff
@@ -0,0 +1,217 @@
...
+    And user "Alice" has created folder "/folderA"
+    And user "Alice" has created folder "/folderB"
+    And user "Alice" has shared folder "/folderA" with user "Brian"
+    And user "Alice" has shared folder "/folderB" with user "Brian"
+    When user "Brian" accepts the following shares offered by user "Alice" using the sharing API
+      | path        |
+      | /folderA    |
+      | /folderB    |
+    And user "Brian" creates folder "/Shares/folderA/ONE" using the WebDAV API
+    And user "Brian" stores id of file "/Shares/folderA/ONE"
```
Comment (kiranparajuli589) [inline tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature:200]: `ONE` is a folder.

**Review by kiranparajuli589 — DISMISSED** [review]

File: tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature (hunk 10 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiWebdavMove2/moveShareOnOcis.feature]: done

File: tests/acceptance/features/apiWebdavMove2/moveFile.feature (hunk 4 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiWebdavMove2/moveFile.feature:405]: done

File: tests/acceptance/features/apiWebdavMove2/moveFileToExcludedDirectory.feature (hunk 7 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiWebdavMove2/moveFileToExcludedDirectory.feature:28]: done

File: tests/acceptance/features/apiWebdavMove2/moveFile.feature (hunk 6 above)
Comment (amrita-shrestha) [reply tests/acceptance/features/apiWebdavMove2/moveFile.feature:458]: done

**Review by kiranparajuli589 — COMMENTED** [review]
please also mention the reasons behind adding a new feature file `moveShareOnOcis.feature`

**Review by kiranparajuli589 — COMMENTED** [review]
lgtm :+1:

**Review by phil-davis — APPROVED** [review]

**Pull request merged**